proknow/Audit.py

```diff
@@ -0,0 +1,159 @@
+__all__ = [
+    'Audit',
+]
+
+import json
+import datetime
+
+from proknow.Exceptions import InvalidOperationError
+
+class Audit(object):
+    """
+    This class should be used to interact with the audit logs in a Proknow organization.
+    """
+
+    def __init__(self, proknow, requestor):
+        """Initializes the Audit class.
+
+        Parameters:
+            proknow (proknow.ProKnow.ProKnow): The ProKnow instance that is instantiating the
+                object.
+            requestor (proknow.Requestor.Requestor): An object used to make API requests.
+        """
+        self._proknow = proknow
+        self._requestor = requestor
+        self.auditResultsPage = None
+    
+    def query(self, page_size=25, **kwargs):
+
+        """Queries for audit logs.
+
+        Parameters:
+            page_size (int, optional): (Default is 25) The number of items for each page
+            start_time (datetime): Start time cut off for whole query
+            end_time (datetime): End time cut off for whole query
+            types (str or list): List of event types
+            user_id (str): ID of events' enacting user
+            user_name (str): Name of events' enacting user
+            patient_id (str): ID of patient
+            patient_name (str): Name of patient
+            patient_mrn (str): Medical record number of patient
+            workspace_id (str): ID of workspace in which events took place
+            workspace_name (str): Name of workspace in which events took place
+            collection_id (str): ID of a collection
+            resource_id (str): ID of a resource
+            resource_name (str): Name of a resource
+            classification (str): 'HTTP' or 'AUTH'
+            methods (str or list): List of HTTP methods: 'GET', 'HEAD', 'POST', 'PUT', 'DELETE', 'CONNECT', 'OPTIONS', 'TRACE', or 'PATCH'
+            uri (str): Filter by the uri that events took place (e.g. '/metrics/custom')
+            user_agent (str): User Agent attributed to events (e.g. Browser User Agent)
+            ip_address (str): IP Address attributed to events
+            status_codes (str or list): List of status codes of events (e.g. 200)
+            text (str or list): Text to search for in all text fields
+
+        Returns:
+            A :class:`proknow.Audit.AuditResultsPage` object, representing a page of query results
```

Comment:
Take a look at some of the methods in Patients.py. The Returns attribute should be of the form `[type]: [description], which allows the Return Type to come through in the documentation.